packages/compass-components/src/components/document-list/element.tsx

```diff
@@ -381,15 +385,14 @@ export const HadronElement: React.FunctionComponent<{
     parentType,
     removed,
     internal,
+    expanded,
+    expand,
+    collapse,
   } = useHadronElement(element);
 
-  useEffect(() => {
-    setExpanded(allExpanded);
-  }, [allExpanded]);
-
   const toggleExpanded = useCallback(() => {
-    setExpanded((val) => !val);
-  }, []);
+    expanded ? collapse() : expand();
+  }, [expand, collapse, expanded]);
```

Comment:
With this deps array, there is really no point in having a use callback here now